cmd/postgres_exporter/postgres_exporter.go

```diff
@@ -653,9 +653,15 @@ func (e *Exporter) checkMapVersions(ch chan<- prometheus.Metric, server *Server)
 	versionDesc := prometheus.NewDesc(fmt.Sprintf("%s_%s", namespace, staticLabelName),
 		"Version string as reported by postgres", []string{"version", "short_version"}, server.labels)
 
+	var semanticVersionShort = semanticVersion.String()
+	if semanticVersion.GE(semver.MustParse("10.0.0")) {
+		var versionSplit = strings.Split(semanticVersion.String(), ".")
```

Comment:
I wonder if checking for version 10 or greater could be fragile. What about always splitting on . but counting the parts instead? That seems safer to me and doesn't depend on the behavior of postgres.